src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -472,6 +474,123 @@ private static Expression AddResponseWritingToMethodCall(Expression methodCall,
             };
         }
 
+        private static MethodInfo GetEnumTryParseMethod()
+        {
+            var staticEnumMethods = typeof(Enum).GetMethods(BindingFlags.Public | BindingFlags.Static)
+                .Where(m => m.IsGenericMethod && m.Name == "TryParse" && m.ReturnType == typeof(bool));
+
+            foreach (var method in staticEnumMethods)
+            {
+                var tryParseParameters = method.GetParameters();
+
+                if (tryParseParameters.Length == 2 &&
+                    tryParseParameters[0].ParameterType == typeof(string) &&
+                    tryParseParameters[1].IsOut)
+                {
+                    return method;
+                }
+            }
+
+            throw new Exception("static bool System.Enum.TryParse<TEnum>(string? value, out TEnum result) does not exist!!?!?");
+        }
+
+        private static MethodInfo GetDateTimeTryPareMethod(Type type)
+        {
+            if (type != typeof(DateTime) && type != typeof(DateOnly) &&
+                 type != typeof(DateTimeOffset) && type != typeof(TimeOnly))
+            {
+                throw new Exception("Parameter is not of type of DateTime, DateOnly, DateTimeOffset, TimeOnly !");
+            }
+
+            var staticDateMethods = type.GetMethods(BindingFlags.Public | BindingFlags.Static)
+                   .Where(m => m.Name == "TryParse" && m.ReturnType == typeof(bool))
+                   .OrderByDescending(m => m.GetParameters().Length);
+
+            foreach (var method in staticDateMethods)
+            {
+                var tryParseParameters = method.GetParameters();
+
+                if (tryParseParameters.Length == 4 &&
+                    tryParseParameters[0].ParameterType == typeof(string) &&
+                    tryParseParameters[1].ParameterType == typeof(IFormatProvider) &&
+                    tryParseParameters[2].ParameterType == typeof(DateTimeStyles) &&
+                    tryParseParameters[3].IsOut &&
+                    tryParseParameters[3].ParameterType == type.MakeByRefType())
+                {
+                    return method;
+                }
+            }
+
+            throw new Exception("static bool TryParse(string?, IFormatProvider, DateTimeStyles, out DateTime result) does not exit!!?!?");
+        }
+
+        // TODO: Use InvariantCulture where possible? Or is CurrentCulture fine because it's more flexible?
+        internal static MethodInfo? FindTryParseMethod(Type type)
+        {
+            static MethodInfo? Finder(Type type)
+            {
+                if (type.IsEnum)
+                {
+                    return EnumTryParseMethod.MakeGenericMethod(type);
+                }
+
+                if (UseTryParseWithDateTimeStyleOptions(type))
+                {
+                    return GetDateTimeTryPareMethod(type);
+                }
+
+                bool useNumberStyle = UseTryParseWithNumberStyleOption(type);
```

Comment:
Let's do something like `bool TryGetNumberStylesTryGetMethod(out MethodInfo? method)`. It's weird to determine we need the `bool TryParse(string, NumberStyles, IFormatProvider, T)` overload but still loop over all the static methods when we could be calling GetMethod() instead.